persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/EntryRecord.java

```diff
@@ -117,17 +150,31 @@ private static int read(FileProvider.Handle handle, ByteBuffer buffer, long posi
       return read;
    }
 
+   private static int arrayLength(byte[] array) {
+      return array == null ? 0 : array.length;
+   }
+
+   private static short arrayLengthAsShort(byte[] array) {
+      return array == null ? 0 : (short) array.length;
+   }
+
+   private static int bufferLength(org.infinispan.commons.io.ByteBuffer byteBuffer) {
+      return byteBuffer == null ? 0 : byteBuffer.getLength();
+   }
+
+   private static short bufferLengthAsShort(org.infinispan.commons.io.ByteBuffer byteBuffer) {
+      return byteBuffer == null ? 0 : (short) byteBuffer.getLength();
+   }
```

Comment:
I don't think these static methods simplify things. `arrayLength` and `bufferLenth` are only called once as well. I would prefer just to keep the checks inline.